lib/plugin/opsgenie/client_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2015-2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opsgenie
+
+import (
+	"context"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateAlert(t *testing.T) {
+	recievedReq := ""
+	testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
```

Comment:
There's a cool lib that allows you to test http requests in a more _realistic_ way:
https://github.com/dnaeon/go-vcr

You can basically set it up to record the request/response (you version those req/resp as yaml files)
The next test run will use that http response when the same request is detected.

We are not using this kind of testing anywhere in Teleport, so just let me know what you think about it even if we don't apply it here.

Reply:
That does seem like a much better way of doing things. Golden file tests like that are usually much nicer to use.